cmake/Dependencies.cmake

```
@@ -726,13 +726,15 @@ if(BUILD_PYTHON)
   endif()
 
   # Check that Python works
+  set(PYTHON_VERSION)
   if(DEFINED PYTHON_EXECUTABLE)
     execute_process(
         COMMAND "${PYTHON_EXECUTABLE}" "--version"
-        RESULT_VARIABLE _exitcode)
+        RESULT_VARIABLE _exitcode OUTPUT_VARIABLE _output_version)
     if(NOT ${_exitcode} EQUAL 0)
       message(FATAL_ERROR "The Python executable ${PYTHON_EXECUTABLE} cannot be run. Make sure that it is an absolute path.")
     endif()
+    string(REGEX MATCH "([0-9]+)\\.([0-9]+)" PYTHON_VERSION ${_output_version})
```

This might fix the CI failure

```suggestion
    string(REGEX MATCH "([0-9]+)\\.([0-9]+)" PYTHON_VERSION "${_python_version}")
```